assets/javascripts/modules/mask.js

```diff
@@ -30,20 +39,72 @@ var maskControlEvent = function ($containerElem) {
   var $control = $containerElem.find('.js-mask-control').first();
   var $publicContent = $containerElem.find('.js-mask-revealed').first();
   var $secretContent = $containerElem.find('.js-mask-secret').first();
+  var $form = $containerElem.find('.js-mask-form').first();
+  var $validationMessage = $containerElem.find('[data-error-message-placeholder]').first();
   var secondsToTimeout = $containerElem.data('mask-timer');
+  var isAuthenticationRequired = $control.data('mask-authentication-required');
 
-  $control.on('click', function(event) {
+  // listen for success event for secure form
+  $control.on('authenticationSuccess', function(event, data) {
     var text = $control.find('[data-toggle-text]').text();
 
-    event.preventDefault();
+    // hide form
+    $form.toggleClass('js-visible').toggleClass('js-hidden');
+
+    // remove any error state on the input
+    $form.find('.form-field').removeClass('form-field--error');
+    $validationMessage.empty();
+
+    // add client secret key
+    $publicContent.text(data);
 
-    toggleState(text, $control, $publicContent, $secretContent);
+    // toggle to reveal client secret
+    toggleValue($publicContent, $secretContent);
 
     // start timer if it's configured
     if(secondsToTimeout) {
       startTimer(text, $control, $publicContent, $secretContent, secondsToTimeout);
     }
+  });
+
+  // listen for success event for secure form
+  $control.on('authenticationFailed', function(error, result) {
```

Comment:
This is specific to the toggled form and shouldn't be necessary as if the callback isn't successful, nothing needs to happen to the mask.